src/components/Expression/AutosizeInput/useTextMetrics.ts

```diff
@@ -1,40 +1,57 @@
-import { MutableRefObject } from "react";
+import { MutableRefObject, useEffect, useRef } from "react";
+
+const canvas = document.createElement("canvas");
+const context = canvas.getContext("2d");
+
+const FONT_PARAMETERS = [
+  "font-style",
+  "font-variant",
+  "font-weight",
+  "font-size",
+  "font-family",
+];
+
+const paddings = (styles?: CSSStyleDeclaration) => {
+  if (!styles) return 0;
 
-const paddings = (styles: CSSStyleDeclaration) => {
   const paddingLeft = parseInt(styles.getPropertyValue("padding-left"), 10);
   const paddingRight = parseInt(styles.getPropertyValue("padding-right"), 10);
 
   return paddingLeft + paddingRight;
 };
 
+const obtainFontString = (styles: CSSStyleDeclaration) => {
+  return FONT_PARAMETERS.reduce(
+    (fontString, cssKey) => `${fontString} ${styles.getPropertyValue(cssKey)}`,
+    ""
+  );
+};
+
+const calculateLength = (text: string) => {
+  if (!context) return 0;
+
+  return context.measureText(text).width;
+};
+
 export const useTextMetrics = (
   elementRef: MutableRefObject<HTMLElement | undefined>
 ) => {
```

Comment:
**suggestion:** it would be nice to have explicit return type here

Reply:
it's inferred, there is no need for that

Reply:
it's inferred, there is no need for that